controllers/quay/quayregistry_controller.go

```diff
@@ -244,10 +244,36 @@ func (r *QuayRegistryReconciler) Reconcile(ctx context.Context, req ctrl.Request
 		}
 	}
 
+	var userProvidedTLSCert []byte
+	if _, ok := configBundle.Data["ssl.cert"]; ok {
+		err = yaml.Unmarshal(configBundle.Data["ssl.cert"], &userProvidedTLSCert)
+		if err != nil {
+			updatedQuay, err = r.updateWithCondition(&quay, v1.ConditionTypeRolloutBlocked, metav1.ConditionTrue, v1.ConditionReasonConfigInvalid, err.Error())
+			if err != nil {
+				log.Error(err, "failed to update `conditions` of `QuayRegistry`")
+
+				return ctrl.Result{}, nil
+			}
+		}
+	}
+	var userProvidedTLSKey []byte
+	if _, ok := configBundle.Data["ssl.key"]; ok {
+		err = yaml.Unmarshal(configBundle.Data["ssl.key"], &userProvidedTLSKey)
+		if err != nil {
+			updatedQuay, err = r.updateWithCondition(&quay, v1.ConditionTypeRolloutBlocked, metav1.ConditionTrue, v1.ConditionReasonConfigInvalid, err.Error())
+			if err != nil {
+				log.Error(err, "failed to update `conditions` of `QuayRegistry`")
+
+				return ctrl.Result{}, nil
+			}
+		}
+	}
+	userProvidedCerts := map[string][]byte{"ssl.key": userProvidedTLSKey, "ssl.cert": userProvidedTLSCert}
+
```

Comment:
This block seems useless. Secret data field is already a `map[string][]byte` and there is no need to unmarshal their content. What is the reasoning behind it?

Reply:
Resolved